core/src/main/java/org/apache/carbondata/core/datastore/filesystem/LocalCarbonFile.java

```diff
@@ -485,4 +486,30 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(file.getAbsolutePath());
   }
+
+  @Override
+  public List<CarbonFile> listDirs() throws IOException {
+    if (!file.isDirectory()) {
+      return new ArrayList<CarbonFile>();
+    }
+    Collection<File> fileCollection = FileUtils.listFilesAndDirs(file,
+            DirectoryFileFilter.DIRECTORY, null);
+    if (fileCollection.isEmpty()) {
+      return new ArrayList<CarbonFile>();
+    }
+    List<CarbonFile> carbonFiles = new ArrayList<CarbonFile>();
+    for (File file : fileCollection) {
+      if (file.isDirectory()) {
+        File[] files = file.listFiles();
```

Comment:
Why list files is needed again? Are you trying to list recursive?

Reply:
try to find all folders from  /tmp/indexservertmp.
no  list recursive, just find the direct directory under /tmp/indexservertmp.

Reply:
This is a generic API.. i think the output for this should be "fileCollection" i.e listing only once...if you want to list the subfolder then better to add a recursive parameter or pass a pattern like /tmp/*/* so that it can list the folders with 2 depth.

Reply:
okay，done